DOC: initialise random number generator before first use in quickstart

shreepads opened the pull request:
numpy's random number generator is used at several places in the quickstart
but a new user, like me, won't know how to initialise it. Therefore I
recommend adding this at the first place rg is used.

<!-- Please be sure you are following the instructions in the dev guidelines
http://www.numpy.org/devdocs/dev/development_workflow.html
-->

<!-- We'd appreciate it if your commit message is properly formatted
http://www.numpy.org/devdocs/dev/development_workflow.html#writing-the-commit-message
-->

shreepads committed:
DOC: initialise random number generator before first use in quickstart

numpy's random number generator is used at several places in the quickstart
but a new user, like me, won't know how to initialise it. Therefore I
recommend adding this at the first place rg is used.

shreepads commented:
Hmm the doc tests have failed as the expected random output doesn't match with that produced. I'm not sure how this test could have been working earlier unless rg is being defined somewhere in the test setup phase. If that seed is known I can incorporate it into the rg initialisation

mattip commented:
The `rng` is initialized in a hidden comment at the top of the file. Perhaps you mean to make that public?

shreepads commented:
> The `rng` is initialized in a hidden comment at the top of the file. Perhaps you mean to make that public?

Right, I didn't see that! Yes I think it should be made public at the first point of use as it tripped me up while I was going thru the quickstart. Let me try adding that.

Just noticed it is also visibly defined lower down in the 'Histogram' section so should be OK here.

shreepads committed:
DOC: make random generator initialisation visible at the first point of use in the quickstart

The random generator in the quickstart is initialised in a hidden test setup
section at the start. This should be made visible at the first point of use
so that new users don't get tripped up trying to set it up. Note that rg is
also visibly initialised lower down in the Histogram section.

rossbar commented on doc/source/user/quickstart.rst:
```diff
@@ -348,6 +347,7 @@ existing array rather than create a new one.
 
 ::
 
+    >>> rg = np.random.default_rng(1)
```
Perhaps a comment could be added here as well, to briefly describe what this line does without distracting from the overall theme of the example (illustrating in-place operations)

shreepads replied:
Makes sense I'll add a comment (formatted like others on the page) - "create instance of default random number generator"

rossbar reviewed (commented):
The proposed change is an improvement - having the instantiation of the Generator instance hidden in the `testsetup` directive would be very confusing for users who are not familiar with the new random API.

On a related note, the quickstart tutorial could probably use some new info/text calling attention to the new random API and [the associated docs](https://numpy.org/devdocs/reference/random/index.html). Just a thought, though it should probably be a separate PR since it's a more involved change.

shreepads committed:
DOC: add inline comment to briefly describe random generator initialisation in quickstart

Clarifies what the statement does without distracting from the overall theme. Also moved
another inline comment in the same example to ensure they are aligned as per the
format used in the quickstart.

WarrenWeckesser commented:
Squashed and merged.  Thanks, @shreepads.

shreepads commented:
Thanks @WarrenWeckesser ! 
I should celebrate my first teensy contribution to a major open source project but there is a pretty strict lockdown here... Let me go to the... kitchen and get... yeah a cup of coffee. :-)
